src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -636,16 +636,30 @@ private void retrievePullRequests(final BitbucketSCMSourceRequest request)
                     } else {
                         branchName = "PR-" + pull.getId() + "-" + strategy.name().toLowerCase(Locale.ENGLISH);
                     }
+                    PullRequestSCMHead head;
+                    if (originBitbucket instanceof BitbucketCloudApiClient) {
+                        head = new PullRequestSCMHead(branchName,
+                                pullRepoOwner,
+                                pullRepository,
+                                repositoryType,
+                                pull.getSource().getBranch().getName(),
+                                pull,
+                                originOf(pullRepoOwner, pullRepository),
+                                strategy
+                        );
+                    } else {
+                        head = new PullRequestSCMHead(branchName,
+                                repoOwner,
+                                repository,
+                                repositoryType,
+                                branchName,
```

Comment:
Here, we could keep the `pull.getSource().getBranch().getName()`, but then, if the `SCMRevision retrieve(SCMHead head, TaskListener listener)` fails, the fallback on finding the commit with the branch name will fail (the branch name is PR-X because of the refspecs)

Reply:
When I tested it, it was with all the strategies for "Discover pull requests from XXX".

I'll give it a second round of test to be sure.

Reply:
You'll need to be sure that the PR repo does not have branch sync enabled, because if bitbucket is keeping the source's "master" branch in lock-step with the target's "master" branch then the merge commit will be buildable from the source repo... but if they have drifted then the PR-merge will not be buildable

Reply:
IOW you need to have Fork Syncing turned off:

<img width="414" alt="screen shot 2017-09-07 at 11 02 24" src="https://user-images.githubusercontent.com/209336/30158124-17b43fc0-93bc-11e7-8931-626ffc0d7053.png">

And then make sure that both repos have divergent commits for the target branch (i.e. add new different commits to the `master` branch of *both* the fork and the target)

Reply:
Don't know why the `master` branch of the fork has anything to do in the merge of a branch of the fork into the `master` branch of the upstream (the target of the PR).
But in my case, as there is no more reference to the fork repo (at least for BitbucketServer), it surely won't have any impact.
Or I'm missing something?

Reply:
Oh I see... you are removing the reference to the fork repo and using the reliance on /refs/pulls/n/head (or whatever it is)... well in that case 👍